File: packages/types/src/block.ts
export type BlockStatus = 'running' | 'success' | 'error' | 'cancelled';
export type OutputType = 'plain' | 'json' | 'diff' | 'image' | 'markdown';

export interface Block {
  id: string;
  session_id: string;
  command: string;
  started_at: number;
  finished_at?: number;
  exit_code?: number;
  status: BlockStatus;
  output_type: OutputType;
  collapsed: boolean;
  ai_explain?: string;
  bookmarked: boolean;
  output_summary?: string;
  anchor_row: number;
  output_row_end: number;
  anchor_y: number;
  output_height_px?: number;
}

File: crates/aurora-core/src/types/block.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub enum BlockStatus {
    Running,
    Success,
    Error,
    Cancelled,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub enum OutputType {
    Plain,
    Json,
    Diff,
    Image,
    Markdown,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Block {
    pub id: String,
    pub session_id: String,
    pub command: String,
    pub started_at: i64,
    pub finished_at: Option<i64>,
    pub exit_code: Option<i32>,
    pub status: BlockStatus,
    pub output_type: OutputType,
    pub collapsed: bool,
    pub ai_explain: Option<String>,
    pub bookmarked: bool,
}
